.github/workflows/schedule_spider.yaml

```diff
@@ -0,0 +1,36 @@
+name: Schedule spider after provided date
+
+on:
+  workflow_dispatch:
+    inputs:
+      spider_name:
+        description: 'Spider to be scheduled'
+        required: true
+      start_date:
+        description: 'Start date (YYYY-MM-DD)'
+        required: true
+
+jobs:
+  schedule:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+    - name: Prepare environment
+      run: |
+        python -m pip install --upgrade pip
+        pip install click python-decouple scrapinghub
+    - name: Schedule job
+      env:
+        SHUB_APIKEY: ${{ secrets.SHUB_APIKEY }}
+        SCRAPY_CLOUD_PROJECT_ID: ${{ secrets.SCRAPY_CLOUD_PROJECT_ID }}
+        FILES_STORE: ${{ secrets.FILES_STORE }}
+        QUERIDODIARIO_DATABASE_URL: ${{ secrets.QUERIDODIARIO_DATABASE_URL }}
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        AWS_ENDPOINT_URL: ${{ secrets.AWS_ENDPOINT_URL }}
+        AWS_REGION_NAME: ${{ secrets.AWS_REGION_NAME }}
+        SPIDERMON_TELEGRAM_SENDER_TOKEN: ${{ secrets.SPIDERMON_TELEGRAM_SENDER_TOKEN }}
+        SPIDERMON_TELEGRAM_RECIPIENTS: ${{ secrets.SPIDERMON_TELEGRAM_RECIPIENTS }}
+        SPIDERMON_TELEGRAM_FAKE: ${{ secrets.SPIDERMON_TELEGRAM_FAKE }}
+      run: python scripts/schedule_job.py ${{ github.event.inputs.spider_name }} --start_date ${{ github.event.inputs.start_date }}
```

Comment:
I believe there's a way to only use only a modified version of this workflow and get the _full crawl_ and _partial crawl_ functionalities using [job envs](https://docs.github.com/en/actions/reference/environment-variables#about-environment-variables) and [steps `if` statements](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsif). Something like:

```suggestion
name: Schedule spider crawl

on:
  workflow_dispatch:
    inputs:
      spider_name:
        description: 'Spider to be scheduled'
        required: true
      start_date:
        description: 'Start date (YYYY-MM-DD)'
        required: false

jobs:
  schedule:
    runs-on: ubuntu-latest
    env:
      SHUB_APIKEY: ${{ secrets.SHUB_APIKEY }}
      SCRAPY_CLOUD_PROJECT_ID: ${{ secrets.SCRAPY_CLOUD_PROJECT_ID }}
      FILES_STORE: ${{ secrets.FILES_STORE }}
      QUERIDODIARIO_DATABASE_URL: ${{ secrets.QUERIDODIARIO_DATABASE_URL }}
      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
      AWS_ENDPOINT_URL: ${{ secrets.AWS_ENDPOINT_URL }}
      AWS_REGION_NAME: ${{ secrets.AWS_REGION_NAME }}
      SPIDERMON_TELEGRAM_SENDER_TOKEN: ${{ secrets.SPIDERMON_TELEGRAM_SENDER_TOKEN }}
      SPIDERMON_TELEGRAM_RECIPIENTS: ${{ secrets.SPIDERMON_TELEGRAM_RECIPIENTS }}
      SPIDERMON_TELEGRAM_FAKE: ${{ secrets.SPIDERMON_TELEGRAM_FAKE }}
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-python@v2
    - name: Prepare environment
      run: |
        python -m pip install --upgrade pip
        pip install click python-decouple scrapinghub
    - name: Schedule full crawl
      if: ${{ !github.event.inputs.start_date }}
      run: python scripts/schedule_job.py ${{ github.event.inputs.spider_name }} --full
    - name: Schedule partial crawl
      if: ${{ github.event.inputs.start_date }}
      run: python scripts/schedule_job.py ${{ github.event.inputs.spider_name }} --start_date ${{ github.event.inputs.start_date }}
```

I'm not really sure if an empty input is evaluated as `false` but then we could check if it's evaluated as `null` instead or setting a `default` for `start_date` to check against.

Reply:
This is exactly what I was trying to do yesterday but I believe I was too tired and didn't understood so well how the `if` works. I tested in my repository and if `start_date` is left empty, it triggers the full crawl.